### PR TITLE
HW/DVD: Refactor DVDInterface and DVDThread to classes.

### PR DESCRIPTION
**Pull request by AdmiralCurtiss**

### CODE REVIEW

File: Source/Core/Core/HW/DVD/DVDInterface.cpp [hunk 1]
```diff
@@ -839,12 +688,11 @@ static bool CheckReadPreconditions()
   return true;
 }
 
-// Iff false is returned, ScheduleEvent must be used to finish executing the command
-static bool ExecuteReadCommand(u64 dvd_offset, u32 output_address, u32 dvd_length,
-                               u32 output_length, const DiscIO::Partition& partition,
-                               ReplyType reply_type, DIInterruptType* interrupt_type)
+// If false is returned, ScheduleEvent must be used to finish executing the command
```
Comment (JosJuice) [inline Source/Core/Core/HW/DVD/DVDInterface.cpp:693]: FYI, `Iff` wasn't a typo. But if you prefer it this way, we can do it this way.

**Review by JosJuice — COMMENTED** [review]
Can't we name the class `DVDInterface` instead of `DVDInterfaceManager`? `Manager` doesn't really add anything to the name.

File: Source/Core/Core/HW/DVD/DVDInterface.cpp (hunk 1 above)
Comment (AdmiralCurtiss) [reply Source/Core/Core/HW/DVD/DVDInterface.cpp:693]: Oh okay, I've never seen that abbreviation before. I'll change it back.

**Comment by JosJuice** [comment]
A side note: When I created DVDThread, I designed it as something that only DVDInterface should have to deal with accessing. So it would be nice to make DVDThread a member of the DVDInterface class instead of the System class... But DI.cpp is currently violating this model, so let's keep this outside the scope of the PR.

**Comment by AdmiralCurtiss** [comment]
> Can't we name the class `DVDInterface` instead of `DVDInterfaceManager`? `Manager` doesn't really add anything to the name.

The reason I added `Manager` for all of these is that I figured having a class be the same name as its enclosing namespace could get both confusing and annoying -- eg. if you want to access `DVDInterface::DriveState` (that is, namespace::enum) from within the `DVDInterface::DVDInterface` (that is, namespace::class) you can't just use `DVDInterface::DriveState` but have to use `::DVDInterface::DriveState`. I do agree that `Manager` doesn't add much but I think it's better to have a different name than the namespace here.

> A side note: When I created DVDThread, I designed it as something that only DVDInterface should have to deal with accessing. So it would be nice to make DVDThread a member of the DVDInterface class instead of the System class... But DI.cpp is currently violating this model, so let's keep this outside the scope of the PR.

Yeah I was wondering about that. That does make sense to change later.

**Comment by JosJuice** [comment]
Right, I hadn't thought about the namespace. What would you think about changing the namespace to e.g. `DVD`?

**Comment by AdmiralCurtiss** [comment]
Sure, I guess we can do it that way.

**Comment by JosJuice** [comment]
Can we do the same for DVDThread? (This will mean it now shares a namespace with DVDInterface, which I think is fine. When they're in separate classes, we no longer need to use the namespace to keep them apart.)

File: Source/Core/Core/System.h [hunk 2]
```diff
@@ -31,13 +31,13 @@ namespace DSP
...
 class DSPManager;
 }
-namespace DVDInterface
+namespace DVD
 {
-class DVDInterfaceState;
+class DVDInterface;
 }
-namespace DVDThread
+namespace DVD
```
Comment (lioncash) [inline Source/Core/Core/System.h:38]: Seems like this can now be merged with the same namespace above

File: Source/Core/Core/HW/DVD/DVDThread.h [hunk 3]
```diff
@@ -32,48 +51,101 @@ class TMDReader;
 class TicketReader;
 }  // namespace IOS::ES
 
-namespace DVDThread
+namespace DVD
```
Comment (lioncash) [inline Source/Core/Core/HW/DVD/DVDThread.h:54]: This can be merged with the above DVD namespace

File: Source/Core/Core/System.h (hunk 2 above)
Comment (AdmiralCurtiss) [reply Source/Core/Core/System.h]: Yup, done.

**Review by JosJuice — APPROVED** [review]

**Review by lioncash — APPROVED** [review]

**Pull request merged**